clients/pkg/promtail/targets/azurelog/parser.go

```diff
@@ -0,0 +1,39 @@
+package azurelog
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/Shopify/sarama"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+type eventHubMessage struct {
+	Records []any `json:"records"`
+}
+
+func messageParser(logger log.Logger, message *sarama.ConsumerMessage) []string {
```

Comment:
WDYT about having the parser return the error vs needing the logger here? If we do keep it this way I think those debug logs further down need to be errors instead.

Reply:
I like the idea 👍 My initial plan was that if there is an error during the parsing, there is not much we can do besides writing the log and ignoring this message in the queue. These errors may occur because, technically, customers can send any data to the event hub.

After your comment, I did think about it again. I like having fewer parameters in the function signature, especially considering there is no need. 

Also, we can introduce error handling: if the message can not be parsed, put the original payload into Loki as a single record. I added these changes to the PR. What do you think?